framework/src/main/java/org/apache/drill/test/framework/CmdConsOut.java

```diff
@@ -3,5 +3,17 @@
 public class CmdConsOut{
   public int exitCode;
   public String consoleOut;
+  public String consoleErr;
   public String cmd;
+
+
+  @Override
+  public String toString() {
+    return "\"CmdConsOut\":{\n" +
+            "\t\"cmd\":\"" + (cmd != null ? cmd : "") + "\",\n" +
+            "\t\"exitCode\":" + exitCode + ",\n" +
+            "\t\"consoleOut\":\"" + (consoleOut != null ? consoleOut : "") + "\",\n" +
+            "\t\"consoleErr\":\"" + (consoleErr != null ? consoleErr : "") + "\"\n" +
+            "}\n";
+  }
 }
```

Comment:
If you need this to be a JSON string, for easier parsing, I think you can create a diff method may be? For the console output (on test results on Jenkins), my opinion is that the current human readable formatted output is better. We could improve it, but I think there may be different requirements for (1) test NG tests and (2) tests results for the conventional tests?

Reply:
My views slightly differ in this aspect
- Current behavior just dumps the output / error after modifying the result returned by command that was run. Instead, it is better to keep ```console output``` and ```console error``` separate.
- For debugging purposes, it is always better to dump the entire state (what command was run, what the exit code was, the console output and console error).
- It is better to dump this information as JSON string. JSON string is also preferred for dumping the information in human readable format.
- In fact, this should be standardized across all objects that contain state information.

Reply:
Sure. Sounds good. Can you also paste an example for regular console output? For error scenarios, I agree with your change though.

I have a few suggestions on the field names to make it more human readable though. Let me know what you think:
- "output":{
	"command":"/root/my-workspace/drill-test-framework/framework/resources/Datasources/tpch/maprdb/json/importTables.sh 1",
	"exitCode":1,
	"output":"", // is this set always? If not we can ignore it if empty?
	"error":"ERROR (22) -  Path specified by '/drill/testdata/tpch_sf1/maprdb/json/customer' does not exist or is not a table.
...
}

Reply:
```CmdConsOut``` status is logged only in 2 places
1. ```DrillTestScript.run()```.
2. ```TestDriver.runGenerateScript()``` (This is currently a debug print only).

An output for a successful ```Utils.execCmd``` (in this case when running ```createViewsJson.sh``` command) is as follows

```"CmdConsOut":{
	"cmd":"/root/my-workspace/drill-test-framework/framework/resources/Datasources/tpcds/createViewsJson.sh",
	"exitCode":0,
	"consoleOut":"
+-------+-------------------------------------------------------+
|  ok   |                        summary                        |
+-------+-------------------------------------------------------+
| true  | Default schema changed to [dfs.tpcds_sf1_json_views]  |
+-------+-------------------------------------------------------+
+-------+-----------------------------------------------------------------------------+
|  ok   |                                   summary                                   |
+-------+-----------------------------------------------------------------------------+
| true  | View 'customer' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+-----------------------------------------------------------------------------+
+-------+----------------------------------------------------------------------------------+
|  ok   |                                     summary                                      |
+-------+----------------------------------------------------------------------------------+
| true  | View 'customer_address' replaced successfully in 'dfs.tpcds_sf1_json_views' schema |
+-------+----------------------------------------------------------------------------------+
+-------+----------------------------------------------------------------------------------+
|  ok   |                                     summary                                      |
+-------+----------------------------------------------------------------------------------+
| true  | View 'customer_demographics' replaced successfully in 'dfs.tpcds_sf1_json_views' schema |
+-------+----------------------------------------------------------------------------------+
+-------+----------------------------------------------------------------------------------+
|  ok   |                                     summary                                      |
+-------+----------------------------------------------------------------------------------+
| true  | View 'household_demographics' replaced successfully in 'dfs.tpcds_sf1_json_views' schema |
+-------+----------------------------------------------------------------------------------+
+-------+-------------------------------------------------------------------------+
|  ok   |                                 summary                                 |
+-------+-------------------------------------------------------------------------+
| true  | View 'item' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+-------------------------------------------------------------------------+
+-------+------------------------------------------------------------------------------+
|  ok   |                                   summary                                    |
+-------+------------------------------------------------------------------------------+
| true  | View 'promotion' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+------------------------------------------------------------------------------+
+-------+-----------------------------------------------------------------------------+
|  ok   |                                   summary                                   |
+-------+-----------------------------------------------------------------------------+
| true  | View 'time_dim' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+-----------------------------------------------------------------------------+
+-------+-----------------------------------------------------------------------------+
|  ok   |                                   summary                                   |
+-------+-----------------------------------------------------------------------------+
| true  | View 'date_dim' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+-----------------------------------------------------------------------------+
+-------+--------------------------------------------------------------------------+
|  ok   |                                 summary                                  |
+-------+--------------------------------------------------------------------------+
| true  | View 'store' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+--------------------------------------------------------------------------+
+-------+--------------------------------------------------------------------------------+
|  ok   |                                    summary                                     |
+-------+--------------------------------------------------------------------------------+
| true  | View 'store_sales' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+--------------------------------------------------------------------------------+
+-------+------------------------------------------------------------------------------+
|  ok   |                                   summary                                    |
+-------+------------------------------------------------------------------------------+
| true  | View 'warehouse' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+------------------------------------------------------------------------------+
+-------+------------------------------------------------------------------------------+
|  ok   |                                   summary                                    |
+-------+------------------------------------------------------------------------------+
| true  | View 'ship_mode' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+------------------------------------------------------------------------------+
+-------+---------------------------------------------------------------------------+
|  ok   |                                  summary                                  |
+-------+---------------------------------------------------------------------------+
| true  | View 'reason' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+---------------------------------------------------------------------------+
+-------+--------------------------------------------------------------------------------+
|  ok   |                                    summary                                     |
+-------+--------------------------------------------------------------------------------+
| true  | View 'income_band' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+--------------------------------------------------------------------------------+
+-------+--------------------------------------------------------------------------------+
|  ok   |                                    summary                                     |
+-------+--------------------------------------------------------------------------------+
| true  | View 'call_center' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+--------------------------------------------------------------------------------+
+-------+-----------------------------------------------------------------------------+
|  ok   |                                   summary                                   |
+-------+-----------------------------------------------------------------------------+
| true  | View 'web_site' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+-----------------------------------------------------------------------------+
+-------+----------------------------------------------------------------------------------+
|  ok   |                                     summary                                      |
+-------+----------------------------------------------------------------------------------+
| true  | View 'store_returns' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+----------------------------------------------------------------------------------+
+-------+-----------------------------------------------------------------------------+
|  ok   |                                   summary                                   |
+-------+-----------------------------------------------------------------------------+
| true  | View 'web_page' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+-----------------------------------------------------------------------------+
+-------+---------------------------------------------------------------------------------+
|  ok   |                                     summary                                     |
+-------+---------------------------------------------------------------------------------+
| true  | View 'catalog_page' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+---------------------------------------------------------------------------------+
+-------+------------------------------------------------------------------------------+
|  ok   |                                   summary                                    |
+-------+------------------------------------------------------------------------------+
| true  | View 'inventory' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+------------------------------------------------------------------------------+
+-------+----------------------------------------------------------------------------------+
|  ok   |                                     summary                                      |
+-------+----------------------------------------------------------------------------------+
| true  | View 'catalog_returns' replaced successfully in 'dfs.tpcds_sf1_json_views' schema |
+-------+----------------------------------------------------------------------------------+
+-------+--------------------------------------------------------------------------------+
|  ok   |                                    summary                                     |
+-------+--------------------------------------------------------------------------------+
| true  | View 'web_returns' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+--------------------------------------------------------------------------------+
+-------+------------------------------------------------------------------------------+
|  ok   |                                   summary                                    |
+-------+------------------------------------------------------------------------------+
| true  | View 'web_sales' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+------------------------------------------------------------------------------+
+-------+----------------------------------------------------------------------------------+
|  ok   |                                     summary                                      |
+-------+----------------------------------------------------------------------------------+
| true  | View 'catalog_sales' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+----------------------------------------------------------------------------------+
",
	"consoleErr":""
}

Reply:
I think your suggestion for readable format looks good. I will make the changes. As per omitting the ```output``` or ```error``` if empty, this can also be done and will be a change in ```toString```.

(Did not omit it in the first place to keep  a consistent schema).

Reply:
Can you please share an updated example?

Reply:
```
"CmdConsOut":{
	"command":"/root/my-workspace/drill-test-framework/framework/resources/Datasources/tpcds/createViewsJson.sh",
	"exitCode":0,
	"output":"
+-------+-------------------------------------------------------+
|  ok   |                        summary                        |
+-------+-------------------------------------------------------+
| true  | Default schema changed to [dfs.tpcds_sf1_json_views]  |
+-------+-------------------------------------------------------+
+-------+-----------------------------------------------------------------------------+
|  ok   |                                   summary                                   |
+-------+-----------------------------------------------------------------------------+
| true  | View 'customer' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+-----------------------------------------------------------------------------+
+-------+----------------------------------------------------------------------------------+
|  ok   |                                     summary                                      |
+-------+----------------------------------------------------------------------------------+
| true  | View 'customer_address' replaced successfully in 'dfs.tpcds_sf1_json_views' schema |
+-------+----------------------------------------------------------------------------------+
+-------+----------------------------------------------------------------------------------+
|  ok   |                                     summary                                      |
+-------+----------------------------------------------------------------------------------+
| true  | View 'customer_demographics' replaced successfully in 'dfs.tpcds_sf1_json_views' schema |
+-------+----------------------------------------------------------------------------------+
+-------+----------------------------------------------------------------------------------+
|  ok   |                                     summary                                      |
+-------+----------------------------------------------------------------------------------+
| true  | View 'household_demographics' replaced successfully in 'dfs.tpcds_sf1_json_views' schema |
+-------+----------------------------------------------------------------------------------+
+-------+-------------------------------------------------------------------------+
|  ok   |                                 summary                                 |
+-------+-------------------------------------------------------------------------+
| true  | View 'item' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+-------------------------------------------------------------------------+
+-------+------------------------------------------------------------------------------+
|  ok   |                                   summary                                    |
+-------+------------------------------------------------------------------------------+
| true  | View 'promotion' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+------------------------------------------------------------------------------+
+-------+-----------------------------------------------------------------------------+
|  ok   |                                   summary                                   |
+-------+-----------------------------------------------------------------------------+
| true  | View 'time_dim' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+-----------------------------------------------------------------------------+
+-------+-----------------------------------------------------------------------------+
|  ok   |                                   summary                                   |
+-------+-----------------------------------------------------------------------------+
| true  | View 'date_dim' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+-----------------------------------------------------------------------------+
+-------+--------------------------------------------------------------------------+
|  ok   |                                 summary                                  |
+-------+--------------------------------------------------------------------------+
| true  | View 'store' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+--------------------------------------------------------------------------+
+-------+--------------------------------------------------------------------------------+
|  ok   |                                    summary                                     |
+-------+--------------------------------------------------------------------------------+
| true  | View 'store_sales' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+--------------------------------------------------------------------------------+
+-------+------------------------------------------------------------------------------+
|  ok   |                                   summary                                    |
+-------+------------------------------------------------------------------------------+
| true  | View 'warehouse' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+------------------------------------------------------------------------------+
+-------+------------------------------------------------------------------------------+
|  ok   |                                   summary                                    |
+-------+------------------------------------------------------------------------------+
| true  | View 'ship_mode' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+------------------------------------------------------------------------------+
+-------+---------------------------------------------------------------------------+
|  ok   |                                  summary                                  |
+-------+---------------------------------------------------------------------------+
| true  | View 'reason' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+---------------------------------------------------------------------------+
+-------+--------------------------------------------------------------------------------+
|  ok   |                                    summary                                     |
+-------+--------------------------------------------------------------------------------+
| true  | View 'income_band' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+--------------------------------------------------------------------------------+
+-------+--------------------------------------------------------------------------------+
|  ok   |                                    summary                                     |
+-------+--------------------------------------------------------------------------------+
| true  | View 'call_center' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+--------------------------------------------------------------------------------+
+-------+-----------------------------------------------------------------------------+
|  ok   |                                   summary                                   |
+-------+-----------------------------------------------------------------------------+
| true  | View 'web_site' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+-----------------------------------------------------------------------------+
+-------+----------------------------------------------------------------------------------+
|  ok   |                                     summary                                      |
+-------+----------------------------------------------------------------------------------+
| true  | View 'store_returns' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+----------------------------------------------------------------------------------+
+-------+-----------------------------------------------------------------------------+
|  ok   |                                   summary                                   |
+-------+-----------------------------------------------------------------------------+
| true  | View 'web_page' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+-----------------------------------------------------------------------------+
+-------+---------------------------------------------------------------------------------+
|  ok   |                                     summary                                     |
+-------+---------------------------------------------------------------------------------+
| true  | View 'catalog_page' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+---------------------------------------------------------------------------------+
+-------+------------------------------------------------------------------------------+
|  ok   |                                   summary                                    |
+-------+------------------------------------------------------------------------------+
| true  | View 'inventory' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+------------------------------------------------------------------------------+
+-------+----------------------------------------------------------------------------------+
|  ok   |                                     summary                                      |
+-------+----------------------------------------------------------------------------------+
| true  | View 'catalog_returns' replaced successfully in 'dfs.tpcds_sf1_json_views' schema |
+-------+----------------------------------------------------------------------------------+
+-------+--------------------------------------------------------------------------------+
|  ok   |                                    summary                                     |
+-------+--------------------------------------------------------------------------------+
| true  | View 'web_returns' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+--------------------------------------------------------------------------------+
+-------+------------------------------------------------------------------------------+
|  ok   |                                   summary                                    |
+-------+------------------------------------------------------------------------------+
| true  | View 'web_sales' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+------------------------------------------------------------------------------+
+-------+----------------------------------------------------------------------------------+
|  ok   |                                     summary                                      |
+-------+----------------------------------------------------------------------------------+
| true  | View 'catalog_sales' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+----------------------------------------------------------------------------------+
",
	"error":""
}

Reply:
(1) Should we use a tab space after the output/error fields?
"CmdConsOut":{
	"command":"/root/my-workspace/drill-test-framework/framework/resources/Datasources/tpcds/createViewsJson.sh",
	"exitCode":0,
	"output":"
                  +-------+-------------------------------------------------------+
                  |  ok   |                        summary                        |
                 +-------+-------------------------------------------------------+
                 | true  | Default schema changed to [dfs.tpcds_sf1_json_views]  |
                 +-------+-------------------------------------------------------+
Not sure if it will work for multi-line, but wanted to see if there was some way to make it consistent. Or may be remove the tab space for above fields? 
(2) Not display "error" or "output" when empty? We could have a toggle - if we need to persist the JSON for Drill to read - only then would the consistent schema help (even there, it shouldn't be a problem)
(3) Rename "CmdConsOut" to "output" or something else more clear?

Reply:
1. The idea is not to modify the output returned from executing the commands itself. The output appears so because that's the output returned from running ```/root/my-workspace/drill-test-framework/framework/resources/Datasources/tpcds/createViewsJson.sh```. 

To add tabs, we have to parse the output and introduce tab to every line that is output. 

My suggestion is to not make this change.

2. Fine. I will make this change. 

3. Name ```output``` will conflict with the field name ```output``` within ```CmdConsOut```. Kept the name of the class since the log output represents the state of an instance of this class.

Reply:
For (1), remove the tabs in your tostring then?

Reply:
Okay

Reply:
Or use two spaces for (1), so that the difference is not jarring? Having the space can make it look like a enclosed block..

Reply:
Also, kindly share a final example, so that we can verify how it looks like.

Reply:
```>> Query: select * from sys.drillbits;
hostname	user_port	control_port	data_port	http_port	current	version	state
qa-node184.qa.lab	31010	31011	31012	8047	false	1.16.0-SNAPSHOT	ONLINE
qa-node188.qa.lab	31010	31011	31012	8047	false	1.16.0-SNAPSHOT	ONLINE
qa-node185.qa.lab	31010	31011	31012	8047	true	1.16.0-SNAPSHOT	ONLINE

> INIT duration: 10.40 s
------------------------------------------------------------------------
> Copying Data
>> Copy duration: 2.704 s

> Generating Data
Running command /root/my-workspace/drill-test-framework/framework/resources/Datasources/tpcds/createViewsJson.sh
"CmdConsOut":{
"command":"/root/my-workspace/drill-test-framework/framework/resources/Datasources/tpcds/createViewsJson.sh",
"exitCode":0,
"output":"
+-------+-------------------------------------------------------+
|  ok   |                        summary                        |
+-------+-------------------------------------------------------+
| true  | Default schema changed to [dfs.tpcds_sf1_json_views]  |
+-------+-------------------------------------------------------+
+-------+-----------------------------------------------------------------------------+
|  ok   |                                   summary                                   |
+-------+-----------------------------------------------------------------------------+
| true  | View 'customer' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+-----------------------------------------------------------------------------+
+-------+----------------------------------------------------------------------------------+
|  ok   |                                     summary                                      |
+-------+----------------------------------------------------------------------------------+
| true  | View 'customer_address' replaced successfully in 'dfs.tpcds_sf1_json_views' schema |
+-------+----------------------------------------------------------------------------------+
+-------+----------------------------------------------------------------------------------+
|  ok   |                                     summary                                      |
+-------+----------------------------------------------------------------------------------+
| true  | View 'customer_demographics' replaced successfully in 'dfs.tpcds_sf1_json_views' schema |
+-------+----------------------------------------------------------------------------------+
+-------+----------------------------------------------------------------------------------+
|  ok   |                                     summary                                      |
+-------+----------------------------------------------------------------------------------+
| true  | View 'household_demographics' replaced successfully in 'dfs.tpcds_sf1_json_views' schema |
+-------+----------------------------------------------------------------------------------+
+-------+-------------------------------------------------------------------------+
|  ok   |                                 summary                                 |
+-------+-------------------------------------------------------------------------+
| true  | View 'item' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+-------------------------------------------------------------------------+
+-------+------------------------------------------------------------------------------+
|  ok   |                                   summary                                    |
+-------+------------------------------------------------------------------------------+
| true  | View 'promotion' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+------------------------------------------------------------------------------+
+-------+-----------------------------------------------------------------------------+
|  ok   |                                   summary                                   |
+-------+-----------------------------------------------------------------------------+
| true  | View 'time_dim' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+-----------------------------------------------------------------------------+
+-------+-----------------------------------------------------------------------------+
|  ok   |                                   summary                                   |
+-------+-----------------------------------------------------------------------------+
| true  | View 'date_dim' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+-----------------------------------------------------------------------------+
+-------+--------------------------------------------------------------------------+
|  ok   |                                 summary                                  |
+-------+--------------------------------------------------------------------------+
| true  | View 'store' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+--------------------------------------------------------------------------+
+-------+--------------------------------------------------------------------------------+
|  ok   |                                    summary                                     |
+-------+--------------------------------------------------------------------------------+
| true  | View 'store_sales' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+--------------------------------------------------------------------------------+
+-------+------------------------------------------------------------------------------+
|  ok   |                                   summary                                    |
+-------+------------------------------------------------------------------------------+
| true  | View 'warehouse' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+------------------------------------------------------------------------------+
+-------+------------------------------------------------------------------------------+
|  ok   |                                   summary                                    |
+-------+------------------------------------------------------------------------------+
| true  | View 'ship_mode' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+------------------------------------------------------------------------------+
+-------+---------------------------------------------------------------------------+
|  ok   |                                  summary                                  |
+-------+---------------------------------------------------------------------------+
| true  | View 'reason' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+---------------------------------------------------------------------------+
+-------+--------------------------------------------------------------------------------+
|  ok   |                                    summary                                     |
+-------+--------------------------------------------------------------------------------+
| true  | View 'income_band' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+--------------------------------------------------------------------------------+
+-------+--------------------------------------------------------------------------------+
|  ok   |                                    summary                                     |
+-------+--------------------------------------------------------------------------------+
| true  | View 'call_center' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+--------------------------------------------------------------------------------+
+-------+-----------------------------------------------------------------------------+
|  ok   |                                   summary                                   |
+-------+-----------------------------------------------------------------------------+
| true  | View 'web_site' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+-----------------------------------------------------------------------------+
+-------+----------------------------------------------------------------------------------+
|  ok   |                                     summary                                      |
+-------+----------------------------------------------------------------------------------+
| true  | View 'store_returns' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+----------------------------------------------------------------------------------+
+-------+-----------------------------------------------------------------------------+
|  ok   |                                   summary                                   |
+-------+-----------------------------------------------------------------------------+
| true  | View 'web_page' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+-----------------------------------------------------------------------------+
+-------+---------------------------------------------------------------------------------+
|  ok   |                                     summary                                     |
+-------+---------------------------------------------------------------------------------+
| true  | View 'catalog_page' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+---------------------------------------------------------------------------------+
+-------+------------------------------------------------------------------------------+
|  ok   |                                   summary                                    |
+-------+------------------------------------------------------------------------------+
| true  | View 'inventory' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+------------------------------------------------------------------------------+
+-------+----------------------------------------------------------------------------------+
|  ok   |                                     summary                                      |
+-------+----------------------------------------------------------------------------------+
| true  | View 'catalog_returns' replaced successfully in 'dfs.tpcds_sf1_json_views' schema |
+-------+----------------------------------------------------------------------------------+
+-------+--------------------------------------------------------------------------------+
|  ok   |                                    summary                                     |
+-------+--------------------------------------------------------------------------------+
| true  | View 'web_returns' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+--------------------------------------------------------------------------------+
+-------+------------------------------------------------------------------------------+
|  ok   |                                   summary                                    |
+-------+------------------------------------------------------------------------------+
| true  | View 'web_sales' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+------------------------------------------------------------------------------+
+-------+----------------------------------------------------------------------------------+
|  ok   |                                     summary                                      |
+-------+----------------------------------------------------------------------------------+
| true  | View 'catalog_sales' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+----------------------------------------------------------------------------------+",
"error":""
}

Reply:
I am okay with no space.

Reply:
Cool. Is (2) yet to be done (blank error / output)?

Reply:
Yes (2) is yet to be done. Needs changes for handling ```,``` towards the end of field.
```,``` has to be included if next field exists, else has to be omitted.

(May be over-engineering for something trivial?)

Reply:
Well, the way I see it, this whole PR is a good to have feature. Most of them are trivial changes meant for better formatting, better parsing, etc.. :) So as much as we'd want to get this in quickly, we also want to do it right when we've spent this much time on it. I hope you agree.

So I think we should handle the empty strings. It's just cleaner to look at from console output.

Reply:
Yup. I agree that the changes have trivial impact on current functionality. Hence in my previous comment I meant to reduce the time and effort in making cosmetic changes :) 

I have pushed the changes for (2)

**ERROR SCENARIO:**
```"CmdConsOut":{
"command":"/root/my-workspace/drill-test-framework/framework/resources/Datasources/tpch/maprdb/json/importTables.sh 1",
"exitCode":1,
"error":"ERROR (22) -  Path specified by '/drill/testdata/tpch_sf1/maprdb/json/customer' does not exist or is not a table.
ERROR (22) -  Path specified by '/drill/testdata/tpch_sf1/maprdb/json/lineitem' does not exist or is not a table.
ERROR (22) -  Path specified by '/drill/testdata/tpch_sf1/maprdb/json/orders' does not exist or is not a table.
ERROR (22) -  Path specified by '/drill/testdata/tpch_sf1/maprdb/json/part' does not exist or is not a table.
ERROR (22) -  Path specified by '/drill/testdata/tpch_sf1/maprdb/json/partsupp' does not exist or is not a table.
ERROR (22) -  Path specified by '/drill/testdata/tpch_sf1/maprdb/json/region' does not exist or is not a table.
ERROR (22) -  Path specified by '/drill/testdata/tpch_sf1/maprdb/json/supplier' does not exist or is not a table.
ERROR (22) -  Path specified by '/drill/testdata/tpch_sf1/maprdb/json/nation' does not exist or is not a table."}
```

**NORMAL SCENARIO:**
```
"CmdConsOut":{
"command":"/root/my-workspace/drill-test-framework/framework/resources/Datasources/tpcds/createViewsJson.sh",
"exitCode":0,
"output":"
+-------+-------------------------------------------------------+
|  ok   |                        summary                        |
+-------+-------------------------------------------------------+
| true  | Default schema changed to [dfs.tpcds_sf1_json_views]  |
+-------+-------------------------------------------------------+
+-------+-----------------------------------------------------------------------------+
|  ok   |                                   summary                                   |
+-------+-----------------------------------------------------------------------------+
| true  | View 'customer' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+-----------------------------------------------------------------------------+
+-------+----------------------------------------------------------------------------------+
|  ok   |                                     summary                                      |
+-------+----------------------------------------------------------------------------------+
| true  | View 'customer_address' replaced successfully in 'dfs.tpcds_sf1_json_views' schema |
+-------+----------------------------------------------------------------------------------+
+-------+----------------------------------------------------------------------------------+
|  ok   |                                     summary                                      |
+-------+----------------------------------------------------------------------------------+
| true  | View 'customer_demographics' replaced successfully in 'dfs.tpcds_sf1_json_views' schema |
+-------+----------------------------------------------------------------------------------+
+-------+----------------------------------------------------------------------------------+
|  ok   |                                     summary                                      |
+-------+----------------------------------------------------------------------------------+
| true  | View 'household_demographics' replaced successfully in 'dfs.tpcds_sf1_json_views' schema |
+-------+----------------------------------------------------------------------------------+
+-------+-------------------------------------------------------------------------+
|  ok   |                                 summary                                 |
+-------+-------------------------------------------------------------------------+
| true  | View 'item' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+-------------------------------------------------------------------------+
+-------+------------------------------------------------------------------------------+
|  ok   |                                   summary                                    |
+-------+------------------------------------------------------------------------------+
| true  | View 'promotion' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+------------------------------------------------------------------------------+
+-------+-----------------------------------------------------------------------------+
|  ok   |                                   summary                                   |
+-------+-----------------------------------------------------------------------------+
| true  | View 'time_dim' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+-----------------------------------------------------------------------------+
+-------+-----------------------------------------------------------------------------+
|  ok   |                                   summary                                   |
+-------+-----------------------------------------------------------------------------+
| true  | View 'date_dim' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+-----------------------------------------------------------------------------+
+-------+--------------------------------------------------------------------------+
|  ok   |                                 summary                                  |
+-------+--------------------------------------------------------------------------+
| true  | View 'store' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+--------------------------------------------------------------------------+
+-------+--------------------------------------------------------------------------------+
|  ok   |                                    summary                                     |
+-------+--------------------------------------------------------------------------------+
| true  | View 'store_sales' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+--------------------------------------------------------------------------------+
+-------+------------------------------------------------------------------------------+
|  ok   |                                   summary                                    |
+-------+------------------------------------------------------------------------------+
| true  | View 'warehouse' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+------------------------------------------------------------------------------+
+-------+------------------------------------------------------------------------------+
|  ok   |                                   summary                                    |
+-------+------------------------------------------------------------------------------+
| true  | View 'ship_mode' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+------------------------------------------------------------------------------+
+-------+---------------------------------------------------------------------------+
|  ok   |                                  summary                                  |
+-------+---------------------------------------------------------------------------+
| true  | View 'reason' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+---------------------------------------------------------------------------+
+-------+--------------------------------------------------------------------------------+
|  ok   |                                    summary                                     |
+-------+--------------------------------------------------------------------------------+
| true  | View 'income_band' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+--------------------------------------------------------------------------------+
+-------+--------------------------------------------------------------------------------+
|  ok   |                                    summary                                     |
+-------+--------------------------------------------------------------------------------+
| true  | View 'call_center' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+--------------------------------------------------------------------------------+
+-------+-----------------------------------------------------------------------------+
|  ok   |                                   summary                                   |
+-------+-----------------------------------------------------------------------------+
| true  | View 'web_site' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+-----------------------------------------------------------------------------+
+-------+----------------------------------------------------------------------------------+
|  ok   |                                     summary                                      |
+-------+----------------------------------------------------------------------------------+
| true  | View 'store_returns' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+----------------------------------------------------------------------------------+
+-------+-----------------------------------------------------------------------------+
|  ok   |                                   summary                                   |
+-------+-----------------------------------------------------------------------------+
| true  | View 'web_page' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+-----------------------------------------------------------------------------+
+-------+---------------------------------------------------------------------------------+
|  ok   |                                     summary                                     |
+-------+---------------------------------------------------------------------------------+
| true  | View 'catalog_page' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+---------------------------------------------------------------------------------+
+-------+------------------------------------------------------------------------------+
|  ok   |                                   summary                                    |
+-------+------------------------------------------------------------------------------+
| true  | View 'inventory' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+------------------------------------------------------------------------------+
+-------+----------------------------------------------------------------------------------+
|  ok   |                                     summary                                      |
+-------+----------------------------------------------------------------------------------+
| true  | View 'catalog_returns' replaced successfully in 'dfs.tpcds_sf1_json_views' schema |
+-------+----------------------------------------------------------------------------------+
+-------+--------------------------------------------------------------------------------+
|  ok   |                                    summary                                     |
+-------+--------------------------------------------------------------------------------+
| true  | View 'web_returns' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+--------------------------------------------------------------------------------+
+-------+------------------------------------------------------------------------------+
|  ok   |                                   summary                                    |
+-------+------------------------------------------------------------------------------+
| true  | View 'web_sales' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+------------------------------------------------------------------------------+
+-------+----------------------------------------------------------------------------------+
|  ok   |                                     summary                                      |
+-------+----------------------------------------------------------------------------------+
| true  | View 'catalog_sales' replaced successfully in 'dfs.tpcds_sf1_json_views' schema  |
+-------+----------------------------------------------------------------------------------+
"}
```

Reply:
Lol, now I see what you were saying. I expected a simpler check, but the code is bit more longer now. 

Regarding the delete char at the end, is that for commas? Last two chars?

Reply:
Every field appends a ```,``` and a ```\n```. So we have to delete both.

Reply:
Got it. Thanks. Makes sense :)